.github/actions/free-hosted-runner-disk-space/action.yml

```diff
@@ -1,18 +1,156 @@
 name: "Free GHA runner disk space"
 description: "Removes unnecessary files from the runner."
 inputs:
+  only-remove-files:
+    description: 'Only remove files and folders, do not manipulate the runner memory'
+    required: false
+    default: 'true'
+  root-reserve-mb:
+    description: 'Space to be left free on the root filesystem, in Megabytes.'
+    required: false
+    default: '12288'
+  temp-reserve-mb:
+    description: 'Space to be left free on the temp filesystem (/mnt), in Megabytes.'
+    required: false
+    default: '100'
+  swap-size-mb:
+    description: 'Swap space to create, in Megabytes.'
+    required: false
+    default: '1024'
+  overprovision-lvm:
+    description: |
+      Create the LVM disk images as sparse files, making the space required for the LVM image files *appear* unused on the
+      hosting volumes until actually allocated. Use with care, this can lead to surprising out-of-disk-space situations.
+      You should prefer adjusting root-reserve-mb/temp-reserve-mb over using this option.
+    required: false
+    default: 'false'
+  build-mount-path:
+    description: 'Absolute path to the mount point where the build space will be available, defaults to /var/lib/docker/ if unset.'
+    required: false
+    default: '/var/lib/docker/'
+  build-mount-path-ownership:
+    description: 'Ownership of the mount point path, defaults to standard "runner" user and group.'
+    required: false
+    default: 'runner:runner'
+  pv-loop-path:
+    description: 'Absolute file path for the LVM image created on the root filesystem, the default is usually fine.'
+    required: false
+    default: '/pv.img'
+  tmp-pv-loop-path:
+    description: 'Absolute file path for the LVM image created on the temp filesystem, the default is usually fine. Must reside on /mnt'
+    required: false
+    default: '/mnt/tmp-pv.img'
```

Comment:
IMO for all these properties the description can be enhanced to include that they're only applicable if `merge-disk-volumes` is `true`